e2e/e2e_test.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2017 The Kubernetes Authors All rights reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/v1"
+)
+
+var _ = Describe("Basic Suite", func() {
+	var helm HelmManager
+	var namespace *v1.Namespace
+	var clientset kubernetes.Interface
+
+	BeforeEach(func() {
+		var err error
+		clientset, err = KubeClient()
+		Expect(err).NotTo(HaveOccurred())
+		By("Creating namespace and initializing test framework")
+		namespaceObj := &v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "e2e-helm-",
+			},
+		}
+		namespace, err = clientset.Core().Namespaces().Create(namespaceObj)
+		Expect(err).NotTo(HaveOccurred())
+		helm = &BinaryHelmManager{
+			Namespace:         namespace.Name,
+			Clientset:         clientset,
+			HelmBin:           helmBinPath,
+			TillerHost:        tillerHost,
+			UseCanary:         true,
+			UseServiceAccount: true,
+		}
+		if !localTiller {
+			Expect(helm.InstallTiller()).NotTo(HaveOccurred())
```

Comment:
Not gonna lie, this syntax reads really weird. It took me a few reads before I realized it was saying "Expect `helm.InstallTiller` to have not failed." I was reading it first as "Expect a Tiller install to not happen"

Reply:
However, the `It` block, is really clean once I understood what was going on

Reply:
It's supposed to follow BDD: https://en.wikipedia.org/wiki/Behavior-driven_development . We can discuss if this test format is a bad match.